src/raft/raft.rs

```diff
@@ -1431,7 +1442,10 @@ impl<T: Storage> Raft<T> {
         }
 
         if send_append {
-            self.send_append(m.get_from());
+            let from = m.get_from();
+            let mut prs = self.take_prs();
```

Comment:
Is it really better than the original implementations?

Reply:
Here it's not better. But `send_append` called in `handle_transfer_leader` needs a `pr` as argument.

Reply:
Can `send_append` accept an `Option`?

Reply:
It's really a good idea? Sometimes it gets `pr` from external arguments, but sometimes from `self.get_prs()`? I think make it unify is also importent for readability.